pints/toy/_german_credit.py

```diff
@@ -0,0 +1,124 @@
+#
+# German credit toy log pdf.
+#
+# This file is part of PINTS.
+#  Copyright (c) 2017-2019, University of Oxford.
+#  For licensing information, see the LICENSE file distributed with the PINTS
+#  software package.
+#
+from __future__ import absolute_import, division
+from __future__ import print_function, unicode_literals
+import numpy as np
+from . import ToyLogPDF
+import io
+import urllib
+import urllib.request
+from scipy import stats
+
+
+class GermanCreditLogPDF(ToyLogPDF):
+    r"""
+    Toy distribution based on a logistic regression model, which takes the
+    form,
+
+    .. math::
+
+        f(x, y|\beta) \propto \text{exp}(-\sum_{i=1}^{N} \text{log}(1 +
+        \text{exp}(-y_i x_i.\beta)) - \beta.\beta/2\sigma^2)
+
+    The data :math:`(x, y)` are a matrix of individual predictors (with 1s in
+    the first column) and responses (1 if the individual should receive credit
+    and -1 if not) respectively; :math:`\beta` is a 25x1 vector of coefficients
+    and :math:`\sigma^2=100`. The dataset here is from [1]_ but the test
+    problem is defined in [2]_.
+
+    Extends :class:`pints.LogPDF`.
+
+    Parameters
+    ----------
+    beta : float
+        vector of coefficients of length 25.
+
+    References
+    ----------
+    .. [1] "UCI machine learning repository", 2010. A. Frank and A. Asuncion.
+           https://archive.ics.uci.edu/ml/datasets/statlog+(german+credit+data)
+    .. [2] "The No-U-Turn Sampler:  Adaptively Setting Path Lengths in
+           Hamiltonian Monte Carlo", 2014, M.D. Hoffman and A. Gelman.
+    """
+    def __init__(self, x=None, y=None, download=False):
```

Comment:
Why do we want to allow x and y to be overridden?

Reply:
Not sure I do? When _both_ (x,y) and download=True are selected then I want to return an error (not overwrite them).